sdk/src/azure/core/az_span.c

```diff
@@ -983,6 +983,34 @@ AZ_NODISCARD AZ_INLINE bool _az_span_url_should_encode(uint8_t c)
   }
 }
 
+AZ_NODISCARD int32_t _az_span_url_encode_calc_length(az_span source)
+{
+  _az_PRECONDITION_VALID_SPAN(source, 0, true);
+
+  int32_t const source_size = az_span_size(source);
+  if (source_size == 0)
+  {
+    return 0;
+  }
+
+  // We may or may not have enough space, given whether the input needs much encoding or not.
```

Comment:
There is no destination. What does it mean by enough space, in this context? I would reword this comment or remove it.